include/zephyr/sys/util.h

```diff
@@ -685,6 +685,16 @@ char *utf8_trunc(char *utf8_str);
  */
 char *utf8_lcpy(char *dst, const char *src, size_t n);
 
+/**
+ * @brief Checks if the given string @p str is UTF-8 encoded.
```

Comment:
It is a bit unclear to me; does this return `false` if `@p str` is not NULL terminated, or does it not consider NULL termination?

Reply:
Oh. I almost forgot about the situation of str for empty. I appreciate your carefulness :)

Reply:
Was not just about the empty string, but in the case that we have maxlen = 4 and str = "abcde", then if we only check the first 4 octets, "abcd" (without the NULL terminator), do we get a false or true return value?

Reply:
To be precise, the variable that affects the process is len, and maxlen is only an external restriction (the reason for this is that 0x00 is also a utf8 character).